### PR TITLE
[Android] Removes dev dependency plugins from release builds

### LINKED ISSUE #157949: Use updated `.flutter-plugins-dependencies` format in the Flutter Gradle plugin
_While strictly speaking it depends on https://github.com/flutter/flutter/issues/157948, this could be started in parallel._

---

In https://github.com/flutter/flutter/issues/157948, we'll add `dev_dependency: true|false` to `.flutter-plugins-dependencies`.

This issue is tracking the Flutter Gradle plugin reading this new key/value pair, and on an _explicit_ value of `true`, excluding a plugin from release branches. I say _explicit_ value as we'll want a bit of backwards-compatibility, so in other words, I'd treat the following scenarios the following way:

```json
{
  "name": "plugin-a",
  "path": "/path/to/plugin-a",
  "dependencies": ["plugin-b", "plugin-c"],
  "native_build": true
}
```

... should imply `dev_dependency: false`.

```json
{
  "name": "plugin-a",
  "path": "/path/to/plugin-a",
  "dependencies": ["plugin-b", "plugin-c"],
  "native_build": true,
  "dev_dependency": false
}
```

... should specify `dev_dependency: false`.

```json
{
  "name": "plugin-a",
  "path": "/path/to/plugin-a",
  "dependencies": ["plugin-b", "plugin-c"],
  "native_build": true,
  "dev_dependency": true
}
```

... should specify `dev_dependency: true`.

### PR DESCRIPTION
**Pull request by camsim99**
Removes plugins that are `dev_dependency`s from being dependencies of release builds of Flutter Android app projects.

Fixes https://github.com/flutter/flutter/issues/157949.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by camsim99** [comment]
plugins_dependencies_test requires an NDK bump: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8730628614122189505/+/u/run_plugin_dependencies_test/stdout

**Comment by camsim99** [comment]
My understanding is that there's no way to get the integration test running in this PR, but if that is not the case please let me know :) I made an attempt to set `presubmit: true`, but I don't think that worked because `bringup` is true?

File: dev/devicelab/bin/tasks/android_release_builds_exclude_dev_dependencies_test.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          await inDirectory(flutterProject.rootPath, () async {
+            await flutter('build', options: <String>[
+              'apk',
+              '--$buildMode',
+              '--target-platform=android-arm',
+            ]);
+
+            final File apk = File('${flutterProject.rootPath}/build/app/outputs/flutter-apk/app-debug.apk');
+            if (!apk.existsSync()) {
+              throw TaskResult.failure("Expected ${apk.path} to exist, but it doesn't");
```
Comment (matanlurey) [inline dev/devicelab/bin/tasks/android_release_builds_exclude_dev_dependencies_test.dart:39]: What is the difference between `throw TaskResult.failure` (here) and `return TaskResult.failure` (below)?

**Review by matanlurey — APPROVED** [review]
LGTM with 2 relatively minor changes/clarifications requested.

Thank you for doing this and being so responsive/incremental!

File: dev/devicelab/bin/tasks/android_release_builds_exclude_dev_dependencies_test.dart (hunk 1 above)
Comment (camsim99) [reply dev/devicelab/bin/tasks/android_release_builds_exclude_dev_dependencies_test.dart:40]: This failure is if the apk that was generated is not found. The one below checks whether or not the dev_dependency plugin was included in the apk or not.

**Pull request merged**